compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -270,6 +271,24 @@ impl<'a> Parser<'a> {
         };
         self.last_unexpected_token_span = Some(self.token.span);
         let mut err = self.struct_span_err(self.token.span, &msg_exp);
+
+        // Add suggestion for a missing closing angle bracket if '>' is included in expected_tokens
+        // there are unclosed angle brackets
+        if self.unmatched_angle_bracket_count > 0
+            && self.token.kind == TokenKind::Eq
+            && expected.iter().any(|tok| match tok {
+                TokenType::Token(t) => *t == TokenKind::Gt,
+                _ => false,
+            })
+        {
+            err.span_suggestion(
+                self.prev_token.span,
+                "maybe try to close unmatched angle bracket",
+                String::new(),
+                Applicability::MaybeIncorrect,
+            );
```

Comment:
This is an inappropriate use of `span_suggestion`. When using that method the third argument is the replacement code that will end up in the given span. In this case what running `rustfix` would do is remove whatever `prev_token` is and _not_ close the angle bracket. If we don't wish to figure out how to properly close (and we likely don't given the difficulty to do so where we could maybe handle a set number of predetermined cases but not the general case), what we _should_ use is a `span_label` instead, which carries no _semantic_ meaning.

Reply:
I used this because I've found it in many different places in the code base, see [here](https://grep.app/search?q=span_suggestion%5C%28%28.%2A%5Cn%29%7B0%2C3%7D.%2AString%3A%3Anew&regexp=true&filter[repo][0]=rust-lang/rust) for all occurrences. Shall I open a PR to fix those too?

Reply:
Those cases _remove_ code at the pointed `Span`, which is why they are being passed an empty `String`. That is a valid use, but the query you made is a good one to run every now and then to audit that no incorrect usages have been introduced :)

In order to check that _future_ changes don't make this mistake we usually create `ui` tests with a comment at the top that says `// run-rustfix` so that the test framework runs `rustfix` and verifies that the resulting code _will_ compile, but we don't have full coverage, particularly for multispan suggestions that aren't currently well supported by our json output.